### DISCUSSION TITLE
How the individual download works

### DISCUSSION THREAD

**Discussion by snow-2357** [shadcn-ui/ui#5401] [Q&A]
How the individual download works , Do u keep everything in different package, if now how this possible, and why more libraries aren't doing it?

**Comment by anishshobithps** [comment]
Are you asking how shadcn implements its individual component download system ? and why other libraries don't adopt to a similar approach ? @snow-2357

**Reply by snow-2357** [reply]
yeah like how they do it, with npx add button how it create a single component  in the the src folder

**Reply by anishshobithps** [reply]
@snow-2357 have a look at https://manupa.dev/blog/anatomy-of-shadcn-ui
